diff --git a/afl/run_all.sh b/afl/run_all.sh
index 142410a..729a33b 100644
--- a/afl/run_all.sh
+++ b/afl/run_all.sh
@@ -1,38 +1,18 @@
 RUNTIME=$1
+RUNTIME="${RUNTIME:-1}"
+(cd csv && sh run_exp.sh $RUNTIME )
+(cd ini && sh run_exp.sh $RUNTIME )
+(cd cjson && sh run_exp.sh $RUNTIME )
+(cd tinyc && sh run_exp.sh $RUNTIME )
+(cd mjs && sh run_exp.sh $RUNTIME )
 
-cd csv
-sh run_exp.sh $RUNTIME &
-cd ..
-cd ini
-sh run_exp.sh $RUNTIME &
-cd ..
-cd cjson
-sh run_exp.sh $RUNTIME &
-cd ..
-cd tinyc
-sh run_exp.sh $RUNTIME &
-cd ..
-cd mjs
-sh run_exp.sh $RUNTIME &
-cd ..
-
-sleep ${RUNTIME}h
+#sleep ${RUNTIME}h
 
 #some safety buffer s.t. all is properly closed and everything had enough time to stop
-sleep 15m
+#sleep 15m
 
-cd csv/eval/
-sh comp_run.sh csvparser &
-cd ../..
-cd ini/eval/
-sh comp_run.sh ini &
-cd ../..
-cd cjson/eval/
-sh comp_run.sh cJSON &
-cd ../..
-cd tinyc/eval/
-sh comp_run.sh tiny &
-cd ../..
-cd mjs/eval/
-sh comp_run.sh mjs &
-cd ../..
+(cd csv/eval/ && sh comp_run.sh csvparser )
+(cd ini/eval/ && sh comp_run.sh ini )
+(cd cjson/eval/ && sh comp_run.sh cJSON )
+(cd tinyc/eval/ && sh comp_run.sh tiny )
+(cd mjs/eval/ && sh comp_run.sh mjs )
diff --git a/klee/run_all.sh b/klee/run_all.sh
index 142410a..34ef6a5 100644
--- a/klee/run_all.sh
+++ b/klee/run_all.sh
@@ -1,38 +1,19 @@
 RUNTIME=$1
+RUNTIME="${RUNTIME:-1}"
+(cd csv && sh run_exp.sh $RUNTIME )
+(cd ini && sh run_exp.sh $RUNTIME )
+(cd cjson && sh run_exp.sh $RUNTIME )
+(cd tinyc && sh run_exp.sh $RUNTIME )
+(cd mjs && sh run_exp.sh $RUNTIME )
 
-cd csv
-sh run_exp.sh $RUNTIME &
-cd ..
-cd ini
-sh run_exp.sh $RUNTIME &
-cd ..
-cd cjson
-sh run_exp.sh $RUNTIME &
-cd ..
-cd tinyc
-sh run_exp.sh $RUNTIME &
-cd ..
-cd mjs
-sh run_exp.sh $RUNTIME &
-cd ..
-
-sleep ${RUNTIME}h
+#sleep ${RUNTIME}h
 
 #some safety buffer s.t. all is properly closed and everything had enough time to stop
-sleep 15m
+#sleep 15m
+
+(cd csv/eval/ && sh comp_run.sh csvparser )
+(cd ini/eval/ && sh comp_run.sh ini )
+(cd cjson/eval/ && sh comp_run.sh cJSON )
+(cd tinyc/eval/ && sh comp_run.sh tiny )
+(cd mjs/eval/ && sh comp_run.sh mjs )
 
-cd csv/eval/
-sh comp_run.sh csvparser &
-cd ../..
-cd ini/eval/
-sh comp_run.sh ini &
-cd ../..
-cd cjson/eval/
-sh comp_run.sh cJSON &
-cd ../..
-cd tinyc/eval/
-sh comp_run.sh tiny &
-cd ../..
-cd mjs/eval/
-sh comp_run.sh mjs &
-cd ../..
diff --git a/pfuzzer/samples/run_all.sh b/pfuzzer/samples/run_all.sh
index 845d317..639c07c 100644
--- a/pfuzzer/samples/run_all.sh
+++ b/pfuzzer/samples/run_all.sh
@@ -1,13 +1,9 @@
 RUNTIME=$1
+RUNTIME="${RUNTIME:-1}"
+(cd ./cjson/ && sh run_exp.sh $RUNTIME cJSON )
+(cd ./ini && sh run_exp.sh $RUNTIME ini )
+(cd ./tinyc && sh run_exp.sh $RUNTIME tiny )
+(cd ./csv && sh run_exp.sh $RUNTIME csvparser )
+(cd ./mjs && sh run_exp.sh $RUNTIME mjs )
+
 
-cd cjson/
-sh run_exp.sh $RUNTIME cJSON &
-cd ../ini
-sh run_exp.sh $RUNTIME ini &
-cd ../tinyc
-sh run_exp.sh $RUNTIME tiny &
-cd ../csv
-sh run_exp.sh $RUNTIME csvparser &
-cd ../mjs
-sh run_exp.sh $RUNTIME mjs &
-cd ..
diff --git a/run_experiments.sh b/run_experiments.sh
index 74b24d3..996aace 100644
--- a/run_experiments.sh
+++ b/run_experiments.sh
@@ -1,7 +1,5 @@
 RUNTIME=$1
-cd afl
-sh run_all.sh $RUNTIME &
-cd ../klee
-sh run_all.sh $RUNTIME &
-cd ../pfuzzer/samples/
-sh run_all.sh $RUNTIME &
+RUNTIME="${RUNTIME:-1}"
+( cd afl && sh run_all.sh $RUNTIME )
+( cd ./klee && sh run_all.sh $RUNTIME )
+( cd ./pfuzzer/samples/ && sh run_all.sh $RUNTIME )
