diff --git a/afl/run_all.sh b/afl/run_all.sh
index 142410a..64b016d 100644
--- a/afl/run_all.sh
+++ b/afl/run_all.sh
@@ -16,10 +16,10 @@ cd mjs
 sh run_exp.sh $RUNTIME &
 cd ..
 
-sleep ${RUNTIME}h
+/bin/sleep ${RUNTIME}h
 
 #some safety buffer s.t. all is properly closed and everything had enough time to stop
-sleep 15m
+/bin/sleep 15m
 
 cd csv/eval/
 sh comp_run.sh csvparser &
diff --git a/klee/run_all.sh b/klee/run_all.sh
index 142410a..64b016d 100644
--- a/klee/run_all.sh
+++ b/klee/run_all.sh
@@ -16,10 +16,10 @@ cd mjs
 sh run_exp.sh $RUNTIME &
 cd ..
 
-sleep ${RUNTIME}h
+/bin/sleep ${RUNTIME}h
 
 #some safety buffer s.t. all is properly closed and everything had enough time to stop
-sleep 15m
+/bin/sleep 15m
 
 cd csv/eval/
 sh comp_run.sh csvparser &
diff --git a/pfuzzer/samples/cjson/run_exp.sh b/pfuzzer/samples/cjson/run_exp.sh
index 5cffd14..c4feaaf 100644
--- a/pfuzzer/samples/cjson/run_exp.sh
+++ b/pfuzzer/samples/cjson/run_exp.sh
@@ -5,7 +5,7 @@ sh clean.sh
 ../../install/bin/trace-instr $(pwd)/${SUBJECT}.c ../excluded_functions > log.txt 2>&1
 cd ../../chains
 time timeout -k9 ${RUNTIME}h python3 chains.py -p ${CURRENTDIR}/${SUBJECT}.c -a False -i False 1> /dev/null 2> error.log &
-sleep ${RUNTIME}h
-sleep 15m
+/bin/sleep ${RUNTIME}h
+/bin/sleep 15m
 cd $CURRENTDIR
 sh get_cov.sh ${SUBJECT} ${CURRENTDIR}
diff --git a/pfuzzer/samples/csv/run_exp.sh b/pfuzzer/samples/csv/run_exp.sh
index 5cffd14..c4feaaf 100644
--- a/pfuzzer/samples/csv/run_exp.sh
+++ b/pfuzzer/samples/csv/run_exp.sh
@@ -5,7 +5,7 @@ sh clean.sh
 ../../install/bin/trace-instr $(pwd)/${SUBJECT}.c ../excluded_functions > log.txt 2>&1
 cd ../../chains
 time timeout -k9 ${RUNTIME}h python3 chains.py -p ${CURRENTDIR}/${SUBJECT}.c -a False -i False 1> /dev/null 2> error.log &
-sleep ${RUNTIME}h
-sleep 15m
+/bin/sleep ${RUNTIME}h
+/bin/sleep 15m
 cd $CURRENTDIR
 sh get_cov.sh ${SUBJECT} ${CURRENTDIR}
diff --git a/pfuzzer/samples/ini/run_exp.sh b/pfuzzer/samples/ini/run_exp.sh
index 5cffd14..c4feaaf 100644
--- a/pfuzzer/samples/ini/run_exp.sh
+++ b/pfuzzer/samples/ini/run_exp.sh
@@ -5,7 +5,7 @@ sh clean.sh
 ../../install/bin/trace-instr $(pwd)/${SUBJECT}.c ../excluded_functions > log.txt 2>&1
 cd ../../chains
 time timeout -k9 ${RUNTIME}h python3 chains.py -p ${CURRENTDIR}/${SUBJECT}.c -a False -i False 1> /dev/null 2> error.log &
-sleep ${RUNTIME}h
-sleep 15m
+/bin/sleep ${RUNTIME}h
+/bin/sleep 15m
 cd $CURRENTDIR
 sh get_cov.sh ${SUBJECT} ${CURRENTDIR}
diff --git a/pfuzzer/samples/mjs/run_exp.sh b/pfuzzer/samples/mjs/run_exp.sh
index 5cffd14..c4feaaf 100644
--- a/pfuzzer/samples/mjs/run_exp.sh
+++ b/pfuzzer/samples/mjs/run_exp.sh
@@ -5,7 +5,7 @@ sh clean.sh
 ../../install/bin/trace-instr $(pwd)/${SUBJECT}.c ../excluded_functions > log.txt 2>&1
 cd ../../chains
 time timeout -k9 ${RUNTIME}h python3 chains.py -p ${CURRENTDIR}/${SUBJECT}.c -a False -i False 1> /dev/null 2> error.log &
-sleep ${RUNTIME}h
-sleep 15m
+/bin/sleep ${RUNTIME}h
+/bin/sleep 15m
 cd $CURRENTDIR
 sh get_cov.sh ${SUBJECT} ${CURRENTDIR}
diff --git a/pfuzzer/samples/tinyc/run_exp.sh b/pfuzzer/samples/tinyc/run_exp.sh
index 5cffd14..c4feaaf 100644
--- a/pfuzzer/samples/tinyc/run_exp.sh
+++ b/pfuzzer/samples/tinyc/run_exp.sh
@@ -5,7 +5,7 @@ sh clean.sh
 ../../install/bin/trace-instr $(pwd)/${SUBJECT}.c ../excluded_functions > log.txt 2>&1
 cd ../../chains
 time timeout -k9 ${RUNTIME}h python3 chains.py -p ${CURRENTDIR}/${SUBJECT}.c -a False -i False 1> /dev/null 2> error.log &
-sleep ${RUNTIME}h
-sleep 15m
+/bin/sleep ${RUNTIME}h
+/bin/sleep 15m
 cd $CURRENTDIR
 sh get_cov.sh ${SUBJECT} ${CURRENTDIR}
